### ISSUE TITLE
Questionnaire Interface Development

### ISSUE THREAD

**Issue by xlai** [xlai/dosefinderhub#5]
Development of the Questionnaire Interface

- [x] Design UI for the questionnaire
- [x] #3 
- [ ] Provide outcome descriptions and explanations
- [x] Process and send user responses to main UI

**Comment by xlai** [comment]
@sdennett20  Let's start by deciding on a simple UI, while you collect responses in relation to #6

**Comment by xlai** [comment]
@sdennett20 Happy for you to draft a documentation and/or guideline on how questions should be added.

Consider the following points:

1. Define a Standardized Question Structure
2. Utilize a Database or Config File for questions
3. Store the questions in a database or a configuration file (e.g. CSV).
4. Create a Question Management Module (optional, to be implemented at later stage)
5. Create functions or modules that can render questions based on the standardized structure
6.  Version Control for Questions 
7. Plan Iterative Reviews

I would suggest a table structure of the following:

order | question_type | question_variable | question_text | option1 | option2 | option3 | ...
-----| -----| -----| -----| -----| -----| -----| -----
1 | mcq |  drug_type | What type of drug are you testing? | chemotherapy | immunotherapy | |
2 | mcq | toxicity_profile | Does the drug have a known toxicity profile? | Yes | No | |
3 | text | trial_objectives | What are the main objectives of the trial?| | | |

N.B. questions are for illustration only

**Comment by sdennett20** [comment]
Hi Xiaoran,
Yesterday me and the guys discussed the questionnaire information and potential questions and I now feel I can get stuck into this issue. I know in my head roughly what I'm aiming for but not really sure how to achieve that in terms of what type of file to use and formatting etc (my lack of computing experience again). Would you mind advising me on this a bit further at some point today? Thanks for the help so far!

**Comment by sdennett20** [comment]
So far I have developed a Questionnaire Guidelines document establishing how we create and record new questions and I have created an initial question database which will be used in the questionnaire UI and passed to the main UI. The latter is subject to change since this afternoon's meeting since @jakeholton is now developing a master list for variable names. This list should be an exhaustive list of variables to be used in the questionnaire UI and will affect how the questions are stored in the question database. I will update this database on receipt of the master list and then begin designing and coding the UI.

We had interesting thoughts about the presentation of user profiles and the metrics we may use to rank one design over another. Issue X delegates the development of these ideas to a time and person in the future, while for now we will keep the user profile and recommendations very simple, probably just a random and arbitrary decision between 3+3 and CRM to get the app up and running.
